src/plugin/ipc/file/ptyconnection.cpp

```diff
@@ -386,6 +386,17 @@ PtyConnection::postRestart()
         controllingTty = jalib::Filesystem::GetControllingTerm(getppid());
       }
       stdinDeviceName = (jalib::Filesystem::GetDeviceName(STDIN_FILENO));
+      if (controllingTty.c_str() == stdinDeviceName ||
```

Comment:
Did you mean it to be `&&` instead of `||`? Unfortunately, I missed it during the original review :(.

Apparently, this commit breaks tcsh, zsh, etc., because the controlling terminal / stdin/out/err are not restored properly.